src/rust/inetstack/protocols/peer.rs

```diff
@@ -132,7 +132,7 @@ impl<N: PhysicalLayer> Peer<N> {
     }
 
     /// Set an SO_* option on the socket.
-    pub fn set_socket_option(&mut self, sd: &mut Socket<N>, option: SocketOption) -> Result<(), Fail> {
+    pub fn set_socket_option(&mut self, sd: &mut Socket, option: SocketOption) -> Result<(), Fail> {
```

Comment:
The naming in this file is a bit confusing. `sd` resonates with `qd` but I don't think there is any direct correlation. If not, then it would be good to call it `socket` or `s`.

Reply:
Ah, the socket descriptor is anything from a file descriptor to an actual data structure, in this case a Socket struct.